spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/TaskValidatorVisitor.java

```diff
@@ -59,65 +64,60 @@ public void reset() {
 		this.recordedSequences.clear();
 		this.transitionsTargetingLabels.clear();
 		this.labelsDefined.clear();
+		this.taskAppNamesWithoutLabels.clear();
 	}
 	
-	public void startVisit(String composedTaskText) {
+	@Override
+	public void startVisit(String composedTaskName, String composedTaskText) {
 		this.composedTaskText = composedTaskText;
 	}
 	
 	@Override
-	public boolean preVisitSequence(LabelledComposedTaskNode firstNode, int sequenceNumber) {
+	public boolean preVisitSequence(LabelledTaskNode firstNode, int sequenceNumber) {
 		if (sequenceNumber > 0 && !firstNode.hasLabel()) {
-			pushProblem(firstNode.getStartPos(), DSLMessage.CT_VALIDATION_SECONDARY_SEQUENCES_MUST_BE_NAMED);
+			pushProblem(firstNode.getStartPos(), DSLMessage.TASK_VALIDATION_SECONDARY_SEQUENCES_MUST_BE_NAMED);
 		}
 		recordedSequences.add(firstNode);
 		return true;
 	}
 
 	@Override
 	public boolean preVisit(SplitNode split) {
+		if (split.getSeriesLength() == 1) {
+			pushProblem(split.startPos,DSLMessage.TASK_VALIDATION_SPLIT_WITH_ONE_FLOW);
+		}
 		if (split.hasLabel()) {
 			String labelString = split.getLabelString();
 			if (labelsDefined.contains(labelString)) {
-				pushProblem(split.getLabel().startPos,DSLMessage.CT_VALIDATION_DUPLICATE_LABEL);
+				pushProblem(split.getLabel().startPos,DSLMessage.TASK_VALIDATION_DUPLICATE_LABEL);
 			}
 			labelsDefined.add(labelString);
 		}
 		return true;
 	}
 
 	@Override
-	public void visit(SplitNode split) {
-		
-	}
-
-	@Override
-	public void postVisit(SplitNode split) {
-	}
-
-	@Override
-	public boolean preVisit(TaskAppNode taskApp) {
-		return true;
-	}
-
-	@Override
 	public void visit(TaskAppNode taskApp) {
 		if (taskApp.hasLabel()) {
 			String labelString = taskApp.getLabelString();
 			if (labelsDefined.contains(labelString)) {
-				pushProblem(taskApp.getLabel().startPos, DSLMessage.CT_VALIDATION_DUPLICATE_LABEL);
+				pushProblem(taskApp.getLabel().startPos, DSLMessage.TASK_VALIDATION_DUPLICATE_LABEL);
 			}
 			labelsDefined.add(labelString);
+			if (taskAppNamesWithoutLabels.contains(labelString)) {
+				pushProblem(taskApp.getLabel().startPos,DSLMessage.TASK_VALIDATION_LABEL_CLASHES_WITH_TASKAPP_NAME);
+			}
+		}
+		else {
+			String name = taskApp.getName();
+			if (labelsDefined.contains(name)) {
+				pushProblem(taskApp.startPos,DSLMessage.TASK_VALIDATION_APP_NAME_CLASHES_WITH_LABEL);
+			}
+			if (taskAppNamesWithoutLabels.contains(name)) {
```

Comment:
Composed Task Runner does not support labels yet.  So this will be one feature (supporting apps with labels) that I'll try to squeeze in between RC and release.